### PR TITLE
fix(scheduler): move JobInfo helper functions to method

### LINKED ISSUE #1341: Move jobInfo help func to member func
**Description**:

Move the following helper functions to member functions of JobInfo.

```
func GetWaitingTime(pg *PodGroup) (*time.Duration, error)
func GetJobPreemptable(pg *PodGroup) bool
func GetJobRevocableZone(pg *PodGroup) string
func GetBudget(pg *PodGroup) *DisruptionBudget
func GetJobMinResources(s scheduling.PodGroupSpec) *api.Resource
```

### PR DESCRIPTION
**Pull request by Thrimbda**
formatted some Doc string

Fixes #1341

### CODE REVIEW

**Comment by Thrimbda** [comment]
Just got in touch with volcano :)
/assign @k82cn PTAL

**Comment by Thrimbda** [comment]
sorry, I'm not very familiar with the e2e tests, but the refactor should not change any logic...

PTAL @k82cn

**Comment by shinytang6** [comment]
> sorry, I'm not very familiar with the e2e tests, but the refactor should not change any logic...
> 
> PTAL @k82cn

There's something wrong with e2e ci, nerver mind. According to the issue, maybe you should also move `GetJobMinResources` to `job_info.go`.
Rest LGTM.

**Comment by Thrimbda** [comment]
sure, I've moved it, PTAL @shinytang6

File: pkg/scheduler/api/job_info.go [hunk 1]
```diff
@@ -336,6 +336,15 @@ func GetBudget(pg *PodGroup) *DisruptionBudget {
 	return NewDisruptionBudget("", "")
 }
 
+// GetJobMinResources return the min resources of podgroup.
+func (ji *JobInfo) GetJobMinResources(s scheduling.PodGroupSpec) *Resource {
```
Comment (k82cn) [inline pkg/scheduler/api/job_info.go:340]: s/GetJobMinResources/GetMinResources

Let's use the PodGroupSpec of JobInfo, so parameter is not necessary.

File: pkg/scheduler/api/job_info.go [hunk 2]
```diff
@@ -239,24 +239,24 @@ func (ji *JobInfo) SetPodGroup(pg *PodGroup) {
...
 			pg.Namespace, pg.Name, err.Error())
 		ji.WaitingTime = nil
 	}
 
-	ji.Preemptable = GetJobPreemptable(pg)
-	ji.RevocableZone = GetJobRevocableZone(pg)
-	ji.Budget = GetBudget(pg)
+	ji.Preemptable = ji.GetJobPreemptable(pg)
+	ji.RevocableZone = ji.GetJobRevocableZone(pg)
+	ji.Budget = ji.GetBudget(pg)
```
Comment (k82cn) [inline pkg/scheduler/api/job_info.go:251]: If it's not used by others, let's make them as private member funcs.

File: pkg/scheduler/api/job_info.go (hunk 1 above)
Comment (Thrimbda) [reply pkg/scheduler/api/job_info.go]: make sense, also I change the signature `s/getWaitingTime/extractWaitingTime` for they are not getters but extract info from another object

File: pkg/scheduler/api/job_info.go (hunk 2 above)
Comment (Thrimbda) [reply pkg/scheduler/api/job_info.go]: sure

**Review by shinytang6 — APPROVED** [review]

**Review by william-wang — APPROVED** [review]
/lgtm
/approve

**Comment by Thrimbda** [comment]
does it require to rebase to merge?

**Comment by k82cn** [comment]
> does it require to rebase to merge?

Waiting for CI happiness :)

**Pull request merged**